android/src/com/mapswithme/maps/editor/data/HoursMinutes.java

```diff
@@ -26,7 +33,21 @@ protected HoursMinutes(Parcel in)
   @Override
   public String toString()
   {
-    return String.format(Locale.US, "%02d:%02d", hours, minutes);
+    final String time = String.format(Locale.US, "%02d:%02d", hours, minutes);
+
+    if (is24HourFormat)
+      return time;
+
+    SimpleDateFormat fmt24 = new SimpleDateFormat("HH:mm");
```

Comment:
Why not to construct a Date object with set hours and minutes, and then directly format it as 12 or 24 hours?